File: test/enum-switch.test.js
const tape = require('tape');
const path = require('path');
const { JVM } = require('../src/jvm');

tape('JVM should handle enum switch statements', async (t) => {
  const jvm = new JVM({ classpath: 'sources' });
  
  // Test the original problematic case
  let output = '';
  jvm.registerJreMethods({
    'java/io/PrintStream': {
      'println(Ljava/lang/String;)V': (jvm, obj, args) => {
        output += args[0] + '\n';
      }
    }
  });

  const classFilePath = path.join('sources', 'EnumSwitchCrash.class');
  
  try {
    await jvm.run(classFilePath);
    t.equal(output.trim(), 'It is red', 'EnumSwitchCrash should print "It is red"');
  } catch (e) {
    t.fail(`EnumSwitchCrash should not crash: ${e.message}`);
  }
  
  // Test comprehensive enum switch
  output = '';
  const testClassFilePath = path.join('sources', 'EnumSwitchTest.class');
  
  try {
    await jvm.run(testClassFilePath);
    const lines = output.trim().split('\n');
    t.equal(lines[0], 'Testing RED: It is red', 'RED case should work');
    t.equal(lines[1], 'Testing GREEN: It is green', 'GREEN case should work');
    t.equal(lines[2], 'Testing BLUE: It is blue', 'BLUE case should work');
    t.equal(lines[3], 'Testing YELLOW: Unknown color', 'Default case should work');
  } catch (e) {
    t.fail(`EnumSwitchTest should not crash: ${e.message}`);
  }

  t.end();
});

File: sources/EnumSwitchTest.java
public class EnumSwitchTest {
    public static void main(String[] args) {
        // Test RED case
        EnumTest.Color color = EnumTest.Color.RED;
        System.out.println("Testing RED: " + testColor(color));
        
        // Test GREEN case  
        color = EnumTest.Color.GREEN;
        System.out.println("Testing GREEN: " + testColor(color));
        
        // Test BLUE case
        color = EnumTest.Color.BLUE;
        System.out.println("Testing BLUE: " + testColor(color));
        
        // Test default case
        color = EnumTest.Color.YELLOW;
        System.out.println("Testing YELLOW: " + testColor(color));

        System.out.println("---");
        System.out.println("Testing Enum.valueOf()...");
        EnumTest.Color red = Enum.valueOf(EnumTest.Color.class, "RED");
        System.out.println("Enum.valueOf(Color.class, RED) = " + red);
    }
    
    public static String testColor(EnumTest.Color color) {
        switch (color) {
            case RED:
                return "It is red";
            case GREEN:
                return "It is green";
            case BLUE:
                return "It is blue";
            default:
                return "Unknown color";
        }
    }
}